interfaces/tests/socket_transport_int/socket_transport_int.c

```diff
@@ -234,4 +253,160 @@ TEST_FUNCTION(send_and_receive_random_buffer_of_random_byte_succeeds)
+    ASSERT_IS_NOT_NULL(chaos_test->listen_socket);
+
+    ASSERT_ARE_EQUAL(int, 0, socket_transport_listen(chaos_test->listen_socket, g_port_num));
+
+    for (i = 0; i < CHAOS_THREAD_COUNT; i++)
+    {
+        ASSERT_ARE_EQUAL(int, 0, socket_transport_connect(chaos_test->client_socket_handles[i], "localhost", g_port_num, 10000));
+
+        chaos_test->incoming_socket_handles[i] = socket_transport_accept(chaos_test->listen_socket);
+
+        ASSERT_IS_NOT_NULL(chaos_test->incoming_socket_handles[i]);
+    }
+
+    (void)interlocked_increment(&chaos_test->API_create_result);
+    wake_by_address_single(&chaos_test->API_create_result);
+    return 0;
+}
+
+/* chaos test
+* This test will do the following:
+* create a bunch of client sockets to connect to binding socket every second
+* create a thread to listen and connect
+* send and receive random buffers of data per client
+* Disconnect a random incoming socket
+* Test if the disconnected socket receives the data from the client socket
+*/
+TEST_FUNCTION(socket_transport_chaos_knight_test)
+{
+    CHAOS_TEST_SOCKETS chaos_knight_test;
+
+    for (size_t i = 0; i < CHAOS_THREAD_COUNT; i++)
+    {
+        chaos_knight_test.client_socket_handles[i] = socket_transport_create_client();
+
+        ASSERT_IS_NOT_NULL(chaos_knight_test.client_socket_handles[i]);
+
+    }
+
+    THREAD_HANDLE thread;
+
+    interlocked_exchange(&chaos_knight_test.API_create_result, 0);
+    ASSERT_ARE_EQUAL(THREADAPI_RESULT, THREADAPI_OK, ThreadAPI_Create(&thread, thread_worker_func, &chaos_knight_test));
+    wait_for_value(&chaos_knight_test.API_create_result, 1);
+
+    for (size_t i = 0; i < CHAOS_THREAD_COUNT; i++)
+    {
+        uint8_t buffer_count = 8;
+
+        SOCKET_BUFFER* send_data;
+        SOCKET_BUFFER* recv_data;
+
+        uint32_t data_size = 64;
+
+        uint32_t total_bytes = make_send_recv_buffer(buffer_count, data_size, &send_data, &recv_data);
+
+        uint32_t bytes_recv;
+        uint32_t bytes_sent;
+        uint32_t flag = SOCKET_SEND_FLAG;
+
+        // Send data back and forth
+        ASSERT_ARE_EQUAL(SOCKET_SEND_RESULT, SOCKET_SEND_OK, socket_transport_send(chaos_knight_test.client_socket_handles[i], send_data, buffer_count, &bytes_sent, flag, NULL));
+        ASSERT_ARE_EQUAL(int, total_bytes, bytes_sent);
+
+        ASSERT_ARE_EQUAL(SOCKET_RECEIVE_RESULT, SOCKET_RECEIVE_OK, socket_transport_receive(chaos_knight_test.incoming_socket_handles[i], recv_data, buffer_count, &bytes_recv, 0, NULL));
+        ASSERT_ARE_EQUAL(int, total_bytes, bytes_recv);
+
+        for (uint32_t inner = 0; inner < buffer_count; inner++)
+        {
+            ASSERT_ARE_EQUAL(int, 0, memcmp(send_data[inner].buffer, recv_data[inner].buffer, send_data[inner].length), "mismatch data from array index %" PRIu32 "", inner);
+
+            free(send_data[inner].buffer);
+            free(recv_data[inner].buffer);
+        }
+        free(send_data);
+        free(recv_data);
```

Comment:
/consider

This code could be in a thread for each and you'd have a bunch of threads sending and receiving then you close one and it should stop and the others should continue for a little bit.

Reply:
You don't have to do this right away, just think about it.